src/commands.ts

```diff
@@ -547,3 +550,162 @@ export function registerCommands(context: vscode.ExtensionContext, prManager: Pu
 		}
 	}));
 }
+
+function sanitizeRepositoryName(value: string): string {
+	return value.trim().replace(/[^a-z0-9_.]/ig, '-');
+}
+
+export function registerGlobalCommands(context: vscode.ExtensionContext, gitAPI: GitAPI, credentialStore: CredentialStore) {
+	// TODO: join with other code from https://github.com/microsoft/vscode-pull-request-github/pull/1661
+	async function getHub(): Promise<GitHub | undefined> {
+		if (await credentialStore.hasOctokit()) {
+			return await credentialStore.getHub()!;
+		}
+
+		const hub = await credentialStore.loginWithConfirmation();
```

Comment:
Ah, I missed this in your other PR, but I think we can simplify this to just calling `credentialStore.login`. Otherwise the behavior is that the user sees a notification asking them to sign in, and if they cancel it, they get a dialog asking  if the extension can sign in. I think we can skip the notification step, this is used just when the extension is starting up